### ISSUE TITLE
Python stub files for torch.nn seem misconfigured

### ISSUE THREAD

**Issue by willfrey** [pytorch/pytorch#31111]
## 🐛 Bug

I'm not receiving any signature or docstring information in VSCode using the Microsoft Python Language Server for the modules defined in the `torch.nn` namespace. Based on `torch/nn/__init__.py`, it appears that the intended design is define classes in submodules of the `torch.nn.modules` namespace and through [this import statement](https://github.com/pytorch/pytorch/blob/master/torch/nn/__init__.py#L1) propagate them up to the `torch.nn` namespace.

The `.pyi` stubs that end up being generated from the `.pyi.in` files for `torch.nn.` import symbols like `from torch.nn.modules.<submodule> import <name> as <name>`, which appears to mask the information declared in the `.pyi` file for `torch.nn.modules.submodule`.

## To Reproduce

1. Open up an empty Python file in VSCode.
1. Select a Python interpreter in VSCode that has `torch` 1.3.0 installed.
1. Ensure that the Microsoft Python Language Server is enabled.
1. Enter the following snippet into an empty Python file in the VSCode editor

```python
from torch.nn import Linear as Linear1
from torch.nn.modules import Linear as Linear2
from torch.nn.modules.linear import Linear as Linear3
```

1. In the same file, begin to type `Linear1()`, `Linear2()`, and `Linear3()`.
1. Observe that for `Linear1()` and `Linear2()` that no type hints or parameter names are provided. They are totally opaque.
1. Notice that for `Linear3()`, parameter names and docstrings are present but no type information appears.

## Expected behavior

I expect the parameter names and docstrings to propagate up to the top level `torch.nn` namespace.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.3.0
 - OS (e.g., Linux): macOS Catalina Version 10.15.2
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): `pip install torch`
 - Python version: 3.7.5
 - CUDA/cuDNN version: N/A
 - GPU models and configuration: N/A
 - Any other relevant information: N/A

## Additional context

For what it's worth, `mypy` does pick up the typed signatures defined in the `.pyi` files. For example, `mypy` will raise an error about incompatible types if I try do the following:

```python

from torch import nn

# mypy will raise the following two errors:
#  Argument 1 to "Linear" has incompatible type "str"; expected "int"
#  Argument 2 to "Linear" has incompatible type "str"; expected "int"
nn.Linear("hello", "there")  
```

cc @vishwakftw @SsnL @jianyuh

**Comment by malmaud** [comment]
I think this is a bug in VSCode's Python language server, not in PyTorch.
